pkg/reconciler/taskrun/taskrun_test.go

```diff
@@ -2067,3 +2067,74 @@ func TestReconcileWorkspaceMissing(t *testing.T) {
 		t.Errorf("Expected TaskRun to fail validation but it did not. Final conditions were:\n%#v", tr.Status.Conditions)
 	}
 }
+
+func TestReconcileTaskResourceResolutionAndValidation(t *testing.T) {
+	for _, c := range []struct {
+		desc string
+		d    test.Data
```

Comment:
could we add a field here like `wantFailedReason` which could be set to `ReasonFailedValidation` or `ReasonFailedResolution`? Then in the comparison we can exactly check if the test case failed for this exact reason!

Reply:
> could we add a field here like wantFailedReason which could be set to ReasonFailedValidation or ReasonFailedResolution? Then in the comparison we can exactly check if the test case failed for this exact reason!

Thanks! it's very nice and make sense.
I fixed it :thumbsup: